sklearn/tree/tree.py

```diff
@@ -303,6 +307,17 @@ def fit(self, X, y, sample_weight=None, check_input=True,
         else:
             min_weight_leaf = 0.
 
+        if not isinstance(beta, float):
+            raise ValueError("beta must be a float")
+        if is_classification:
+            if not 0. <= beta <= 1.:
+                raise ValueError("beta must be in range [0., 1.] "
```

Comment:
It is true that classification shouldn't be above 1.0, but entropy has a stricter bound depending on the number of classes. I can't remember if they are reweighted to scale to 1.0 though? It might be better to just take in a positive number and let users figure it out.